### PR TITLE
store frequently used server addresses; fixes #20

### LINKED ISSUE #20: DST-GR-0050: Store frequently used servers
The software shall provide UI and means to store frequently used server addresses.

Two usages of the UserPreferences Service are needed right now
- [ ] HubController connection Login dialog => DEHPCommon
- [ ] OpcClient connection dialog => DEHPEcosimPro

The DEHPEcosimPro will handle user preferences that won't be shared across other dst adapter. So It needs it's own implementation of the user preferences by extending the common user preference class.

### PR DESCRIPTION
**Pull request by ahmedatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/DEHP-ECOSIMPRO/pulls) open
- [x] I have verified that I am following the DEHP-EcosimPro [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/DEHP-ECOSIMPRO/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Added the AppSettings class, which derives from the UserPreference class and contains a set of locally stored
Using the UserPreferenceService in the DstLoginViewModel to read and update a set of locally stored server addresses.
Changed the DstLogin view ServerAddress input from a TextBox to a ComboBoxEdit, having its data source as a list of URIs saved locally.

### CODE REVIEW

**Review by alexatrhea — APPROVED** [review]

File: DEHPEcosimPro/Settings/AppSettings.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace DEHPEcosimPro.Settings
+{
+    using System.Collections.Generic;
+
+    using DEHPCommon.UserPreferenceHandler;
+
+    public class AppSettings : UserPreference
```
Comment (nathanatrhea) [inline DEHPEcosimPro/Settings/AppSettings.cs:31]: Missing doc

File: DEHPEcosimPro/Settings/AppSettings.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        public List<string> SavedOpcUris { get; set; }
```
Comment (nathanatrhea) [inline DEHPEcosimPro/Settings/AppSettings.cs:33]: Missing doc

File: DEHPEcosimPro/Settings/AppSettings.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+        public AppSettings()
```
Comment (nathanatrhea) [inline DEHPEcosimPro/Settings/AppSettings.cs:35]: Missing doc

File: DEHPEcosimPro/ViewModel/Dialogs/DstLoginViewModel.cs [hunk 4]
```diff
@@ -123,12 +130,31 @@ public bool RequiresAuthentication
...
+
+        /// <summary>
+        /// Backing field for <see cref="SavedUris"/>
+        /// </summary>
+        private ReactiveList<string> savedUris;
+
+        /// <summary>
+        /// Gets or sets the saved server addresses
+        /// </summary>
+        public ReactiveList<string> SavedUris
```
Comment (nathanatrhea) [inline DEHPEcosimPro/ViewModel/Dialogs/DstLoginViewModel.cs:142]: ReactiveList don't need that the private corresponding field is explicitly implemented

**Review by nathanatrhea — CHANGES_REQUESTED** [review]

**Pull request merged**